R/estimateAlpha.R

```diff
@@ -0,0 +1,274 @@
+                           "log_modulo_skewness_diversity", "shannon_diversity"),
+           "dominance" = c("absolute_dominance",
+                           "dbp_dominance", "core_abundance_dominance",
+                           "gini_dominance", "dmn_dominance", "relative_dominance",
+                           "simpson_lambda_dominance"),
+           "evenness" = c("camargo_evenness", "pielou_evenness", "simpson_evenness",
+                          "evar_evenness", "bulla_evenness"),
+           "richness" = c("ace_richness", "chao1_richness", "hill_richness", 
+                          "observed_richness"))
+}
+
+.alpha_rarefaction <- function(
+        x,
+        n.iter=1L,
+        args.sub=list(assay.type="counts", min_size=min(colSums(assay(x, "counts")),
+                                                        na.rm = TRUE),
+                      verbose=FALSE),
+        FUN=.estimate_diversity,
+        args.fun=c(index="shannon",
+                   assay.type="subsampled"),
+        ...,
+        name = args.fun$index) {
+    # Calculating the mean of the subsampled alpha estimates ans storing them
+    colData(x)[, name] <- lapply(seq(n.iter), function(j){
+        # subsampling the counts from the original tse object
+        x_sub <- do.call(subsampleCounts, args = c(list(x), args.sub))
+        # calculating the diversity indices on the subsampled object
+        # warnings are supressed due to the depricated warning of the alpha
+        # measure functions
+        suppressWarnings(x_sub <- do.call(FUN, args = c(list(x_sub),
+                                                        args.fun,
+                                                        list(...))))
+        # Storing estimate results
+        colData(x_sub)[, args.fun$index, drop=FALSE]
+    }) %>% data.frame(.) %>% rowMeans(.) %>% data.frame(.)
+    return(x)
+}
+
+.parse_name <- function(
+        index, name, measure) {
+    # parsing name string to use as a column name at colData when storing estimates
+    if (name==index) {
+        # check if suffix of the alpha indices if present at index
+        # otherwise keeping suffix as a name if name not defined by user.
+        if (measure %in% unlist(strsplit(index, "\\_"))) {
+            name <- index
+        } else {
+            name <- paste0(index, "_", measure)
+        }
+    } else {
+        # don't change name if defined by user
+        return(name)
+    }
+}
+
+.estimate_diversity <- function(
+        x, assay.type = "counts",
+        index = c("coverage", "fisher", "gini_simpson",  "inverse_simpson",
+                  "log_modulo_skewness","shannon"),
+        name = index, ...) {
+    estimateDiversity(x, assay.type=assay.type, index=index, name=name, ...)
+}
+
+.estimate_dominance <- function(
+        x,
+        assay.type = "counts", 
+        index = c("absolute", "dbp", "core_abundance", "gini", "dmn",  "relative",
+                  "simpson_lambda"),
```

Comment:
Move these functions under estimate*.R file. Move the functionality of estimateDiversity under .estimate_diversity. --> Then we can use these functions in the future internally without warnings

Reply:
Easiest and tidiest way is to just rename estimateDiversity with .estimate_diversity and create a new exported function estimateDiversity which utilizes this "old" function

Cannot see if you have already made changes because Im on my phone